### ISSUE TITLE
Plugins Sprint 194/195

### ISSUE THREAD

**Issue by ericwill** [eclipse-che/che#18529]
## Sprint 194/195 (December 9th 2020 to January 19th 2021)
This sprint will result in the release of Che 7.24, and possibly Che 7.25.

### Epics
* Allow Che 7 Git + SSH Flow #13494 @vitaliy-guliy 
    * [ ] Check for credentials (SSH) before trying to clone git repositories: https://github.com/eclipse/che/issues/13494#issuecomment-694387267

### CRW
* [ ] [CONTENT_UPDATE] AG - Customizing the devfile and plug-in registries documentation overly hard to read and understand [RHDEVDOCS-1762](https://issues.redhat.com/browse/RHDEVDOCS-1762) @ericwill 
* [x] [EUG] Create content for 'What is a devfile' section about 'A devfile as a a separate, locally stored artifact' [RHDEVDOCS-2300](https://issues.redhat.com/browse/RHDEVDOCS-2300) @ericwill 
* [x] Update documentation of artifact repositories using untrusted TLS certificates [RHDEVDOCS-2311](https://issues.redhat.com/browse/RHDEVDOCS-2311) @tsmaeder 
    * [x] Update documentation of artifact repositories using untrusted TLS certificates #18131
* [x] Consider adding VS Code GitHub Pull Request plug-in to CRW 2.6 [CRW-1226](https://issues.redhat.com/browse/CRW-1226) @vinokurig 
    * [x] EUG - [DOWNSTREAMING] Remove doc re: VS Code Github Pull Request plugin (as it's not in CRW) [RHDEVDOCS-2224](https://issues.redhat.com/browse/RHDEVDOCS-2224)
* [ ] VS Code extensions providing code snippets APIs make not usable in CRW [CRW-1294](https://issues.redhat.com/browse/CRW-1294) @tsmaeder 
    * [ ] Java code snippets 'try_catch' and 'main' are not working #16502
    * [ ] Snippets don't work when Plugin Host is not colocated with Back End eclipse-theia/theia#8460
* [x] Use oc binary from oc tools rpm instead of kubectl binary from google [CRW-1401](https://issues.redhat.com/browse/CRW-1401) @ericwill 
* [x] Use 3rd party icons in CRW plugin registry instead of invisble ones [CRW-1385](https://issues.redhat.com/browse/CRW-1385) @ericwill
* [x] The "Deploy to OpenShift" command failed in the "RedHatFuse" workspace [CRW-1409](https://issues.redhat.com/browse/CRW-1409) @svor 
* [x] URLs of the project sample are not established after running the "2. Start Quarkus in devmode"in the "Quarkus" workspace [CRW-1411](https://issues.redhat.com/browse/CRW-1411) @svor 
    * [x] Endpoints don't work in Quarkus workspaces #18512
* [x] Failed to deploy sample to the Openshift cluster from the "NodeJS Configmap" based workspace [CRW-1412](https://issues.redhat.com/browse/CRW-1412) @svor
* [x] redhat/java plugin requires to declare a volume for dependencies stored in m2 [CRW-1418](https://issues.redhat.com/browse/CRW-1418) @svor 
* [ ] Support EAP XP 2.0 [CRW-1419](https://issues.redhat.com/browse/CRW-1419) @svor 
* [x] remove unneeded node / update to node 12 in sidecars [CRW-1421](https://issues.redhat.com/browse/CRW-1421)

### Bugs
* [x] [UX] When clicking "Run" nothing happens unless a java file is open / extension is loaded #18004 @sunix 
* [x] Asciidoc export to pdf fails #16370 @sunix 
* [ ] Add resource limits/requests to the devfiles and plugins in the registries #16685
    * Revisit eclipse/che-plugin-registry#471: only do [CRW plugins](https://github.com/redhat-developer/codeready-workspaces/tree/crw-2.6-rhel-8/dependencies/che-plugin-registry/v3/plugins)
    * Investigate limits/requests for the devfile registry
* [x] Importing several private projects in a single devfile fails #18494 @sunix 
* [x] Check if the GitHub oAuth is available when asking to sign in the Pull Requests plugin #18237 @vinokurig 
    * This item is blocked by a che-server/dashboard issue

### Unplanned
* [x] Failure to clone zip projects when using OpenShift trusted CA bundle #18654 @vitaliy-guliy 
* [ ] github pr plugin: fork creation is hanging when creating a pull request #18664 @vinokurig 

### Open PRs From Last Sprint
* [x] Workspace Resource Monitor Plugin #17205 @svor 

#### Theia API
* [x] Apply CommentController plugin API eclipse-theia/theia#8492 @vinokurig 

#### Automation / Tests
* [x] Create a che-editors file to store all editors plugin definitions #18214 @ericwill 
* [ ] Ability to run tests of a vscode extension on top of che-theia #18219 @svor 

#### Maintenance
* [ ] Check extensions of VS Code extension pack are running smoothly #15390 @sunix 
* Plugin Updates
    * [ ] Update netcoredbg plugin #18462 @svor 
    * [ ] Update sonarlint-vscode to 1.17.0 eclipse/che-plugin-registry#609 @sunix 
* [ ] Rework che-theia build process to not use source linking #16852 @tsmaeder 

#### Availability
| Name  | Availability | 0.6x |
| ------------- | ------------- | ------------- |
| @vitaliy-guliy  | 28  | 17 |
| @vinokurig  | 22  | 13 |
| @ericwill | 15 | 9 |
| @tsmaeder | 19 | 11 |
| @svor | 27 | 16 |
| @sunix | 24 | 14 |

##### @ericwill
| Priority  | Item |
| ------------- | ------------- |
| 1 |  Create a che-editors file to store all editors plugin definitions #18214 |
| 2 | [CONTENT_UPDATE] AG - Customizing the devfile and plug-in registries documentation overly hard to read and understand [RHDEVDOCS-1762](https://issues.redhat.com/browse/RHDEVDOCS-1762) |
| 3  | EUG] Create content for 'What is a devfile' section about 'A devfile as a a separate, locally stored artifact' [RHDEVDOCS-2300](https://issues.redhat.com/browse/RHDEVDOCS-2300)  |
| 4 | Use oc binary from oc tools rpm instead of kubectl binary from google [CRW-1401](https://issues.redhat.com/browse/CRW-1401) |
| 5 | Use 3rd party icons in CRW plugin registry instead of invisble ones [CRW-1385](https://issues.redhat.com/browse/CRW-1385) |

##### @tsmaeder 
| Priority  | Item |
| ------------- | ------------- |
| 1  |  VS Code extensions providing code snippets APIs make not usable in CRW [CRW-1294](https://issues.redhat.com/browse/CRW-1294) |
| 2 | Update documentation of artifact repositories using untrusted TLS certificates [RHDEVDOCS-2311](https://issues.redhat.com/browse/RHDEVDOCS-2311) |
| 3 | Rework che-theia build process to not use source linking #16852 |

##### @svor 
| Priority  | Item |
| ------------- | ------------- |
| 1  | The "Deploy to OpenShift" command failed in the "RedHatFuse" workspace [CRW-1409](https://issues.redhat.com/browse/CRW-1409) |
| 2  | URLs of the project sample are not established after running the "2. Start Quarkus in devmode"in the "Quarkus" workspace [CRW-1411](https://issues.redhat.com/browse/CRW-1411) |
| 3 | Failed to deploy sample to the Openshift cluster from the "NodeJS Configmap" based workspace [CRW-1412](https://issues.redhat.com/browse/CRW-1412) |
| 4 | redhat/java plugin requires to declare a volume for dependencies stored in m2 [CRW-1418](https://issues.redhat.com/browse/CRW-1418) |
| 5 | remove unneeded node / update to node 12 in sidecars [CRW-1421](https://issues.redhat.com/browse/CRW-1421) |
| 6 | Support EAP XP 2.0 [CRW-1419](https://issues.redhat.com/browse/CRW-1419) |
| 7 | Ability to run tests of a vscode extension on top of che-theia #18219 |
| 8 | Update netcoredbg plugin #18462 |

##### @vinokurig  
| Priority  | Item |
| ------------- | ------------- |
| 1  |  Apply CommentController plugin API eclipse-theia/theia#8492 |
| 2  | Consider adding VS Code GitHub Pull Request plug-in to CRW 2.6 [CRW-1226](https://issues.redhat.com/browse/CRW-1226)  |
| 3 | Check if the GitHub oAuth is available when asking to sign in the Pull Requests plugin #18237 |

##### @vitaliy-guliy   
| Priority  | Item |
| ------------- | ------------- |
| 1 | Failure to clone zip projects when using OpenShift trusted CA bundle #18654 |
| 2  | Add resource limits/requests to the devfiles and plugins in the registries #16685 |
| 3 | Check for credentials (SSH) before trying to clone git repositories: https://github.com/eclipse/che/issues/13494#issuecomment-694387267 |

##### @sunix
| Priority  | Item |
| ------------- | ------------- |
| 1 | Asciidoc export to pdf fails #16370 |
| 2 | The rust getting started sample should use the vscode extension #17069  |
| 3 | Importing several private projects in a single devfile fails #18494 |
| 4 | [UX] When clicking "Run" nothing happens unless a java file is open / extension is loaded #18004 |
| 5 | Check extensions of VS Code extension pack are running smoothly #15390 |
| 6 | Update sonarlint-vscode to 1.17.0 #609 |

**Comment by benoitf** [comment]
As reported during prio call, please remove the line 'Rework che-theia build process to not use source linking #16852'
Also it's in a bug section while it's not even a bug
This enhancement is not even requested by PM/Architects or highly voted by people

**Comment by ericwill** [comment]
> As reported during prio call, please remove the line 'Rework che-theia build process to not use source linking #16852'
> Also it's in a bug section while it's not even a bug

You're right, it's not a bug, and I have moved it to the maintenance section.

> This enhancement is not even requested by PM/Architects or highly voted by people

It has been prioritized at the bottom of the queue for this sprint. Just because PM/Architects didn't request does not mean it cannot be worked on. As I understand it, @tsmaeder would like this enhancement to make his development flow easier. As long as PM/Architect priorities are being worked on, I don't see why we cannot work on this as well.

**Comment by ericwill** [comment]
Superseded by #18808
